semana10/astrotrips/src/containers/App/header.js

```diff
@@ -0,0 +1,190 @@
+    ]
+  }}
+
+  render() {
+    const { classes } = this.props;
+    return (
+      <div className={classes.root}>
+        <AppBar position="static">
+          <Toolbar>
+            <IconButton 
+              edge="start" 
+              className={classes.menuButton} 
+              color="inherit" 
+              aria-label="menu"
+              onClick={()=> this.handleDrawer()}
+            >
+              <MenuIcon />
+            </IconButton>
+            <Typography 
+              variant="h6" className={classes.title} 
+              style={{ cursor: "pointer" }}
+              onClick={() => this.props.goToHome()}
+            >
+              FutureX
+            </Typography>
+            {this.props.isLogged && (
+              <div>
+                <IconButton
+                  aria-label="account of current user"
+                  aria-controls="menu-appbar"
+                  aria-haspopup="true"
+                  onClick={this.handleMenu}
+                  color="inherit"
+                >
+                  <AccountCircle />
+                </IconButton>
+                <Menu
+                  id="menu-appbar"
+                  anchorEl={this.state.anchorEl}
+                  anchorOrigin={{
+                    vertical: 'top',
+                    horizontal: 'right',
+                  }}
+                  transformOrigin={{
+                    vertical: 'top',
+                    horizontal: 'right',
+                  }}
+                  open={this.state.openMenu}
+                  onClose={this.handleClose}
+                  style={{marginTop: "36px"}}
+                >
+                  <MenuItem onClick={this.handleMenu}>Close</MenuItem>
+                  <Divider/>
+                  <MenuItem onClick={(e) => {
+                    this.handleMenu(e)
+                    this.props.doLogout()
+                  }}>
+                    Logout
+                  </MenuItem>
+                </Menu>
+              </div>
+            )}
+            {
+              !this.props.isLogged &&
+              <Button 
+                variant="outlined" 
+                color="secondary"
+                onClick={this.props.goToLogin}
+                size="small"
+              >
+                Admin
+              </Button>
+            }
+          </Toolbar>
+        </AppBar>
+        <Drawer
+          className={classes.drawer}
+          variant="persistent"
+          anchor="left"
+          open={this.state.openDrawer}
+          classes={{
+            paper: classes.drawerPaper,
+          }}
+        >
+          <div className={classes.drawerHeader}>
+            <List>
+              <ListItem key={1}>
+                <ListItemIcon>
+                  <FlightTakeoff />
+                </ListItemIcon>
+                <ListItemText primary={"Welcome!"} />
+              </ListItem>
+            </List>
+            <IconButton onClick={() => this.handleDrawer()}>
```

Comment:
Aqui não era necessára a criação dessa arrow function, poderia ser só `onClick={this.handleDrawer}`